core/src/test/java/org/apache/iceberg/TestScanPlanningAndReporting.java

```diff
@@ -236,9 +241,9 @@ public void scanningWithSkippedDataFiles() throws IOException {
     assertThat(result.resultDeleteFiles().value()).isEqualTo(0);
     assertThat(result.scannedDataManifests().value()).isEqualTo(1);
     assertThat(result.scannedDeleteManifests().value()).isEqualTo(0);
-    assertThat(result.skippedDataManifests().value()).isEqualTo(1);
+    assertThat(result.skippedDataManifests().value()).isEqualTo(2);
```

Comment:
The overwrite rewrites the original manifest and adds a new one for the new data file.